### PR TITLE
fix(mongodb_metrics source, postgresql_metrics source): remove connect in SourceConfig::build

### PR DESCRIPTION
**Pull request by fanatid**
Part of #6086

### CODE REVIEW

File: src/sources/mongodb_metrics/mod.rs [hunk 1]
```diff
@@ -273,13 +196,23 @@ impl MongoDBMetrics {
...
-            .context(CommandBuildInfoMongoError)?;
-        from_document(doc).context(CommandBuildInfoParseError)
+            .map_err(CollectError::Mongo)?;
+        from_document(doc).map_err(CollectError::Bson)
+    }
+
+    async fn print_version(&self) -> Result<(), CollectError> {
+        let node_type = self.get_node_type().await?;
+        let build_info = self.get_build_info().await?;
+        debug!(
```
Comment (jszwedko) [inline src/sources/mongodb_metrics/mod.rs:212]: Given this message is only printed at the debug level, would it make sense to not even fetch the node type and build info if debug logs are not enabled?

I'm also fine with dropping this log message in-lieu of debug messages during the actual collection.

**Review by jszwedko — APPROVED** [review]
Just one note about whether we even care to fetch the server metadata during collection simply for the debug log. Otherwise this looks great! Thanks for the clean-up.

File: src/sources/mongodb_metrics/mod.rs (hunk 1 above)
Comment (fanatid) [reply src/sources/mongodb_metrics/mod.rs]: Updated for mongo and postgres :slightly_smiling_face:

**Review by jszwedko — APPROVED** [review]
Thanks @fanatid !

**Pull request merged**